ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -1318,9 +1313,16 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
         // Find the index segment corresponding to the token indexStartOffset.
         // Get entries starting from the token Key in this index.
         startTimeInMs = time.milliseconds();
-        StoreFindToken newToken =
-            findEntriesFromSegmentStartOffset(storeToken.getOffset(), storeToken.getStoreKey(), messageEntries,
-                new FindEntriesCondition(maxTotalSizeOfEntries), indexSegments, storeToken.getInclusive());
+        StoreFindToken newToken;
+        //if index based token point to the last entry of last index segment, and journal size cleaned up after last log
+        //segment auto close, we should return the original index based token instead of getting from journal.
```

Comment:
I also wonder how index-based token would point to last index segment in normal case. Is it because previous journal-based token gets converted to index-based after last log segment is closed?

Reply:
Discussed offline and point to the right place where the journal based token transformed to index based token.